### ISSUE TITLE
Parser(query).tables not working correctly with some queries after 2.7.0 update

### ISSUE THREAD

**Issue by JPBou** [macbre/sql-metadata#514]
## Issue with Parsing Tables in `sql-metadata` Version 2.7.0 and Later

In versions of `sql-metadata` up to 2.6.0, queries such as the one below parsed tables correctly:

```python
from sql_metadata import Parser

query = """
SELECT o.column_1, e.column_2
FROM test_schema.test_table o
LEFT JOIN test_schema_2.test_table_2 AS e
    ON o.column_1 = replace(e.column_2, chr(10), '\n')
"""
print(Parser(query).tables)
# Output: ['test_schema.test_table', 'test_schema_2.test_table_2']
```

However, starting from version 2.7.0, the same query results in an incorrect parsing of tables:
```python
from sql_metadata import Parser

query = """
SELECT o.column_1, e.column_2
FROM test_schema.test_table o
LEFT JOIN test_schema_2.test_table_2 AS e
    ON o.column_1 = replace(e.column_2, chr(10), '\n')
"""
print(Parser(query).tables)
# Output: ['test_schema.test_table', 'test_schema_2.test_table_2', 'chr']
```

**Comment by MiuNice** [comment]
After conducting a basic investigation, I discovered that the issue stems from the _determine_last_relevant_keyword function in the parser.py file.
```python
if token.value == "," and token.last_keyword_normalized == "ON":
    return "FROM"
```
This appears to be the cause.

**Comment by MiuNice** [comment]
> 经过基本调查后，我发现问题源于 parser.py 文件中的 _determine_last_relevant_keyword 函数。
> 
> ```python
> if token.value == "," and token.last_keyword_normalized == "ON":
>     return "FROM"
> ```
> 
> 这看起来就是原因。

Modified #422

**Comment by MiuNice** [comment]
#422

**Comment by macbre** [comment]
Resolved by #423.
